### PR TITLE
Csg/feature/get-pase-data-via-rpc

### LINKED ISSUE #25: Implement PASE hooks
#### Problem
Need hooks into the SDK to get the PASE data out

 #### Proposed Solution
<suggested fix, suggested enhancement>
Add code to trace the PASE session and send the data out via rpc

### PR DESCRIPTION
**Pull request by wahajsyed**
Problem
Implement hooks to trace and send PASE data

Summary of Changes
RPC to fetch PASE data
Changes to the SDK to trace and save the PASE data
Fixes #25

### CODE REVIEW

File: src/controller/python/chip/ChipDeviceCtrl.py [hunk 1]
```diff
@@ -311,4 +327,3 @@ def _InitLib(self):
 
             self._dmLib.pychip_GetCommandSenderHandle.argtypes = [c_void_p]
             self._dmLib.pychip_GetCommandSenderHandle.restype = c_uint64
```
Comment (wahajsyed) [inline src/controller/python/chip/ChipDeviceCtrl.py:329]:
```suggestion
            self._dmLib.pychip_GetCommandSenderHandle.restype = c_uint64

```

File: src/csg_test_harness/utilities.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+std::string PAKE_1_key ("PAKE1");
+std::string PAKE_1_Pa_key ("PAKEPa");
+std::string PAKE_1_key_id_key ("PAKEKeyID");
+
+std::string PAKE_2_key ("PAKE2");
+std::string PAKE_2_encryption_id_key ("EncryptionID");
+
+std::string PAKE_3_key ("PAKE3");
+
+const char * yaml_string_for_map(std::map<std::string, std::map<std::string, std::string>> *trace_map)
```
Comment (kidip) [inline src/csg_test_harness/utilities.cpp:43]: I wonder if there is a library or something that does this for us? Worried if there are any edge cases that might bite us.

**Review by kidip — APPROVED** [review]

File: src/csg_test_harness/utilities.cpp (hunk 2 above)
Comment (wahajsyed) [reply src/csg_test_harness/utilities.cpp:43]: Filed https://github.com/chip-csg/connectedhomeip/issues/28 to investigate

Comment (wahajsyed) [reply src/csg_test_harness/utilities.cpp:43]: https://github.com/jbeder/yaml-cpp/blob/master/LICENSE

File: src/csg_test_harness/utilities.cpp [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            // for each message item
+            result_yaml_str += "  "; // indent 2 spaces for nested hash
+            result_yaml_str += y.first;
+            result_yaml_str += ": ";
+            result_yaml_str += y.second;
+            result_yaml_str += "\n";
+        }
+    }
+
+    char * response = new char [result_yaml_str.length()+1];
```
[the remark this answers is not in the document]
Comment (wahajsyed) [reply src/csg_test_harness/utilities.cpp:60]: We need to return a char * here so we can stick with supported ctypes primitives std:string and c++ data types are not supported.
You are right, we need to figure out a way to free this memory

**Comment by wahajsyed** [comment]
New PR: https://github.com/chip-csg/connectedhomeip/pull/30/files

**Comment by wahajsyed** [comment]
@cjandhyala  - Added a todo to free the memory in the other PR, due to time sensitivity will address the issue in a later pr

**Comment by wahajsyed** [comment]
Closing this PR, with newer PR: https://github.com/chip-csg/connectedhomeip/pull/30/files

**Pull request closed without merging**